tests/transforms/test_tape_expand.py

```diff
@@ -388,7 +388,7 @@ class NonDiffPhaseShift(qml.PhaseShift):
 
 
 # Custom decomposition functions for testing.
-def custom_cnot(wires):
+def custom_cnot(wires, **_):
```

Comment:
`tape_expand` will pass the hyperparameters of an operation to the decomposition function. For `CNOT`, now that it inherits from `Controlled`, it inherits the hyperparameters such as `control_wires` and `control_values`, which all gets passed into the this function.

Reply:
we might want to try to fix this.. but it's not the highest priority today given the hackiness of custom decomps as it is. If we can document this, it wouldn't hurt, plus we need to update existing docs:
https://github.com/PennyLaneAI/pennylane/blob/6a537027b31ceaa85751cbaba63d555c8425264f/pennylane/__init__.py#L298
https://github.com/PennyLaneAI/pennylane/blob/6a537027b31ceaa85751cbaba63d555c8425264f/doc/introduction/compiling_circuits.rst?plain=1#L239

Reply:
I updated the two docs you mentioned.